File: python/sdk_quick_start.py
"""
SDK quick start — autonomous first-time flow.

Run:
    pip install pathcourse-sdk
    python python/sdk_quick_start.py \
        --tx 0xYOUR_DEPOSIT_TX \
        --wallet 0xYOUR_SENDING_WALLET

What this does:
1. Calls pathcourse.claim_key(tx_hash, wallet) to retrieve an API key for
   your on-chain deposit. Polls up to ~3 min while the gateway reconciles
   your USDC transfer on Base.
2. Constructs a PathCourseClient with the returned key.
3. Makes a single pch-fast inference call so you can verify the full loop.

If you already have a saved API key and just want to test the client,
set PCH_API_KEY instead and skip --tx / --wallet.
"""

import argparse
import os
import sys

import pathcourse
from pathcourse import PathCourseClient, PCH_FAST


def main() -> int:
    parser = argparse.ArgumentParser()
    parser.add_argument("--tx",     help="Base L2 tx hash of your USDC deposit to the PCH treasury")
    parser.add_argument("--wallet", help="Sending wallet address (0x...)")
    args = parser.parse_args()

    api_key = os.environ.get("PCH_API_KEY")
    if not api_key:
        if not args.tx or not args.wallet:
            print("Provide PCH_API_KEY, or both --tx and --wallet.", file=sys.stderr)
            return 1
        print(f"Claiming API key for deposit {args.tx[:10]}... ", flush=True)
        result = pathcourse.claim_key(tx_hash=args.tx, wallet=args.wallet)
        api_key = result["api_key"]
        print(f"  tier     : {result.get('tier')}")
        print(f"  balance  : {result.get('balance_usdc')} USDC")
        print(f"  agent_id : {result.get('agent_id')}")
        print("  Save this key — it is only shown once.")

    client = PathCourseClient(api_key=api_key)
    assert client.verify_key(), "API key rejected by gateway"
    print("verify_key -> OK\n")

    resp = client.chat(
        messages=[{"role": "user", "content": "Reply with exactly: SDK smoke test OK"}],
        model=PCH_FAST,
        max_tokens=20,
    )
    print(f"model    : {resp.model}")
    print(f"response : {resp.content!r}")
    print(f"usage    : {resp.usage}")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: javascript/sdk_quick_start.js
/**
 * SDK quick start — autonomous first-time flow.
 *
 * Run:
 *     npm install @pathcourse/sdk
 *     node javascript/sdk_quick_start.js --tx 0x... --wallet 0x...
 *
 * What this does:
 * 1. Calls claimKey({ tx_hash, wallet }) to retrieve an API key for your
 *    on-chain deposit. Polls up to ~3 min while the gateway reconciles
 *    your USDC transfer on Base.
 * 2. Constructs a PathCourseClient with the returned key.
 * 3. Makes a single pch-fast inference call so you can verify the full loop.
 *
 * If you already have a saved API key, set PCH_API_KEY and skip --tx/--wallet.
 */

import { claimKey, PathCourseClient, PCH_FAST } from '@pathcourse/sdk';

function argFlag(name) {
  const i = process.argv.indexOf(`--${name}`);
  return i >= 0 ? process.argv[i + 1] : null;
}

async function main() {
  let apiKey = process.env.PCH_API_KEY;

  if (!apiKey) {
    const tx     = argFlag('tx');
    const wallet = argFlag('wallet');
    if (!tx || !wallet) {
      console.error('Provide PCH_API_KEY, or both --tx and --wallet.');
      process.exit(1);
    }
    console.log(`Claiming API key for deposit ${tx.slice(0, 10)}...`);
    const r = await claimKey({ tx_hash: tx, wallet });
    apiKey = r.api_key;
    console.log(`  tier     : ${r.tier}`);
    console.log(`  balance  : ${r.balance_usdc} USDC`);
    console.log(`  agent_id : ${r.agent_id}`);
    console.log('  Save this key — it is only shown once.');
  }

  const client = new PathCourseClient({ apiKey });
  if (!(await client.verifyKey())) throw new Error('API key rejected by gateway');
  console.log('verify_key -> OK\n');

  const resp = await client.chat({
    messages:  [{ role: 'user', content: 'Reply with exactly: SDK smoke test OK' }],
    model:     PCH_FAST,
    maxTokens: 20,
  });
  console.log('model    :', resp.model);
  console.log('response :', JSON.stringify(resp.content));
  console.log('usage    :', resp.usage);
}

main().catch((err) => { console.error(err); process.exit(1); });
